dfusion_rust_core/src/models/account_state.rs

```diff
@@ -20,6 +20,15 @@ pub struct AccountState {
 }
 
 impl AccountState {
+    pub fn default(state_index: U256) -> Self {
```

Comment:
State index is relevant information here. I suppose we could get away without it (since the caller knows the batch_index they are calling with), but It would be like intentionally forgetting relevant information.